create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java

```diff
@@ -0,0 +1,98 @@
+package no.sikt.nva.scopus;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import nva.commons.core.Environment;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ScopusS3ClientTest {
+
+    public static final String PREFIX_FILENAME_1 = "prefixFilename1";
+    public static final String PREFIX_FILENAME_2 = "prefixFilename2";
+    public static final String FAILING = "I am failing!";
+    private final String bucketName = "bucketName";
+    private final String filename = "filename";
+    private final String prefix = "prefix";
+    private AmazonS3 amazonS3Client;
+    private Environment environment;
+
+    @BeforeEach
+    public void setUp() {
+        amazonS3Client = mock(AmazonS3.class);
+        environment = mock(Environment.class);
+        when(environment.readEnv(ScopusS3Client.BUCKET_NAME)).thenReturn(bucketName);
+        when(environment.readEnv(ScopusS3Client.AWS_REGION)).thenReturn("eu-west-1");
+    }
+
+    @Test
+    public void testConstructorsForCodeCoverage() {
+        ScopusS3Client client = new ScopusS3Client(environment);
+    }
+
+    @Test
+    public void testGetFile() {
+        S3Object s3object = mock(S3Object.class);
```

Comment:
We have a FakeS3Client that can be used to test the functionality of communicating with S3

Reply:
In which package do you get access to that FakeS3Client? And why not just mcok it?

Reply:
Usually when we mock something, we use Mockito's any() methods. This means that your test will return the "expected" file even when the path you are requesting is wrong. FakeS3Client enables you to just store a file and then retrieve it as you would have done with the normal s3 client. But this is just a suggestion. The fake clients are in the nva-testutils package